src/app/pages/datasets/components/dataset-form/sections/other-options-section/other-options-section.component.ts

```diff
@@ -207,6 +208,11 @@ export class OtherOptionsSectionComponent implements OnInit, OnChanges {
       return;
     }
 
+    let specialSmallBlockSize = getFieldValue(this.existing.special_small_block_size, this.parent) as (number | 'INHERIT');
+    if (specialSmallBlockSize !== 'INHERIT') {
```

Comment:
It seems you are only interested in whether value was inherited. It's worth extracting relevant code from `valueOrInherit` into a new method.